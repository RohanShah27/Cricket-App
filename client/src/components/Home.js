import React, { Component } from "react";
import { connect } from "react-redux";
import "../index.css";
import dhoni from "../DHONI.png";
import pakistan from "../assests/pakistan.jpg";
import india from "../assests/india.jpg";
import { getHeadlines, getHeadline } from "../actions/Headlines";
import { getMatchesRecentMatches } from "../actions/matchActions";
import { getFixtures } from "../actions/Fixtures";

export class Home extends Component {
  constructor(props) {
    super(props);
  }
  componentWillMount() {
    this.props.getMatchesRecentMatches();
    this.props.getHeadline(1);
    this.props.getHeadlines();
    this.props.getFixtures();
  }
  render() {
    console.log("check props home", this.props);
    return (
      <div style={{ marginTop: "80px" }}>
        <div className="home-flex-container" id="test">
          <div className="site">
            <div className="fixtures ">
              <h2 className="fixtures-header" id="fixtures-header">
                Fixtures
              </h2>
              <div className="col-content">
                <ul className="home-ul">
                  {this.props.fixtures.map((index, fixture) => (
                    <p class="home-p">
                      <span id={"matchName"}>{fixture.match}</span>{" "}
                      <span className="home-time">
                        {fixture.match_date} at {fixture.time}
                      </span>
                      <hr className="home-hr" />
                    </p>
                  ))}
                </ul>
              </div>
            </div>
            {/* News Section */}
            <div className="news">
              <div className="col-center">
                <h2 className="news-header" id="news-Section">
                  India's Tour of WestIndies
                </h2>
                <div col-content>
                  <figure>
                    <img src={dhoni} className="homeimg" />
                  </figure>
                  <figcaption>
                    <span className="newsTitle">
                      {this.props.headline[0].headlines}
                    </span>
                    <hr className="home-hr" />
                    <p
                      className="home-p"
                      style={{ fontSize: "15px", paddingLeft: "5px" }}
                    >
                      {/* Single Headline */}
                      {this.props.headline[0].headlines_description}
                    </p>
                  </figcaption>
                </div>
              </div>
            </div>
            <div className="headlines">
              <h2 className="fixtures-header" id="headlines">
                Headlines
              </h2>
              <div className="col-content">
                <ul className="home-ul">
                  {this.props.headlines.map(single_headline => (
                    <>
                      <a
                        onClick={() =>
                          this.props.getHeadline(single_headline.headline_id)
                        }
                      >
                        {" "}
                        <li className="home-li" style={{ cursor: "pointer" }}>
                          {single_headline.headlines}
                        </li>
                      </a>
                      <hr className="home-hr" />
                    </>
                  ))}
                </ul>
              </div>
            </div>
            <div className="home-match-container">
              <div className="testimonials" id="#recent-matches">
                {this.props.match.map(match => (
                  <div className="card">
                    <span className="tournament-home">
                      {" "}
                      {match.competition == "others"
                        ? "International"
                        : match.competition}{" "}
                      - {match.match_type}
                    </span>

                    <div className="parent">
                      <div className="first">
                        <img className="imgformatch" src={india} />
                        <p style={{ wordBreak: "break-word" }}>{match.team1}</p>
                      </div>
                      <div className="second">
                        <button className="buttonformatch">View</button>
                      </div>
                      <div className="third">
                        <img className="imgformatch" src={pakistan} />

                        <p>{match.team2}</p>
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          </div>
        </div>{" "}
      </div>
    );
  }
}
const mapStateToProps = state => ({
  headlines: state.headlinesReducer.headlines,
  headline: state.headlinesReducer.headline,
  fixtures: state.fixturesReducer.fixtures,
  match: state.matchReducers.match
});
export default connect(
  mapStateToProps,
  { getHeadlines, getFixtures, getHeadline, getMatchesRecentMatches }
)(Home);
