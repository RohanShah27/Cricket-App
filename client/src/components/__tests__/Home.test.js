import React from "react";
import { shallow, mount } from "enzyme";
import { Home } from "../Home";
const history = { push: jest.fn() };

let fixtures = [
  {
    team1_name: "India",
    team2_name: "Pakistan",
    match_date: "29/11/1997",
    match_time: "12:00 IST",
    match_type: "ODI"
  },
  {
    team1_name: "India",
    team2_name: "Pakistan",
    match_date: "29/11/1997",
    match_time: "12:00 IST",
    match_type: "ODI"
  }
];
let headlines = [
  { headline_id: 1, headlines: "Master Blaster Sachin baby" },
  { headline_id: 2, headlines: "BCCI President elections coming up" }
];
let headline = [
  {
    headlines: "Test headline",
    headline_description: "test Descrption"
  }
];
let match = [
  {
    match_id: 2,
    match_competition: "Big bAsh leagues",
    match_type: "ODI",
    team1: "India",
    team2: "Pakistan"
  },
  {
    match_id: 3,
    match_competition: "Big bAsh leagues",
    match_type: "T20",
    team1: "India",
    team2: "Srilanka"
  }
];
const home = jest.fn();
const getFixtures = jest.fn();
const getMatchesRecentMatches = jest.fn();
const getHeadline = jest.fn();
const getHeadlines = jest.fn();
const wrapper = mount(
  <Home
    home={home}
    getFixtures={getFixtures}
    getMatchesRecentMatches={getMatchesRecentMatches}
    getHeadline={getHeadline}
    getHeadlines={getHeadlines}
    fixtures={fixtures}
    headlines={headlines}
    match={match}
    headline={headline}
    history={history}
  />
);

describe("Test MAtch Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should check for component did mount as a function", () => {
    const componentWillMount = jest.spyOn(Home.prototype, "componentWillMount");
    wrapper.instance().componentWillMount();
    expect(componentWillMount).toHaveBeenCalled();
  });
  it("should have a container with heading called fixtures", () => {
    expect(wrapper.find("#fixtures-header")).toBeTruthy();
    console.log(wrapper.find("#fixtures-header"));
  });
  it("should have a container with heading called fixtures", () => {
    expect(wrapper.find("#fixtures-header").props().children).toBe("Fixtures");
  });
  it("should have a container with heading called news", () => {
    expect(wrapper.find("#news-Section").props().children).toBe("News");
  });
  it("should have a container with heading called HEadlines", () => {
    expect(wrapper.find("#headlines").props().children).toBe("Headlines");
  });
  it("should have a match name in fixture ", () => {
    expect(wrapper.find("#matchName0").props().children).toStrictEqual([
      fixtures[0].team1_name,
      " VS ",
      fixtures[0].team2_name,
      " -",
      " ",
      fixtures[0].match_type
    ]);
  });
  it("should have a match timings in fixture ", () => {
    expect(wrapper.find("#matchTimings0").props().children).toStrictEqual([
      fixtures[0].match_date,
      " at ",
      fixtures[0].match_time
    ]);
  });
  it("should have a match name in fixture ", () => {
    expect(wrapper.find("#matchName1").props().children).toStrictEqual([
      fixtures[1].team1_name,
      " VS ",
      fixtures[1].team2_name,
      " -",
      " ",
      fixtures[1].match_type
    ]);
  });
  it("should have a match timings in fixture ", () => {
    expect(wrapper.find("#matchTimings1").props().children).toStrictEqual([
      fixtures[1].match_date,
      " at ",
      fixtures[1].match_time
    ]);
  });
  it("should have a headline in headlines ", () => {
    expect(wrapper.find("#singleheadline0").props().children).toBe(
      headlines[0].headlines
    );
  });
  it("should check for headline in news section", () => {
    expect(wrapper.find("#headlineTitle").text()).toBe("");
  });
  it("should check for headline description in news section", () => {
    expect(wrapper.find("#headlineDescription").text()).toBe(
      headline[0].headlines
    );
  });
  it("should have a match team one in matches ", () => {
    expect(wrapper.find("#matchTeamOne0").props().children).toBe(
      match[0].team1
    );
  });
  it("should have a match team two in matches ", () => {
    expect(wrapper.find("#matchTeamTwo0").props().children).toBe(
      match[0].team2
    );
  });
  it("should have a match team one in matches ", () => {
    expect(wrapper.find("#matchTeamOne1").props().children).toBe(
      match[1].team1
    );
  });
  it("should have a match team two in matches ", () => {
    expect(wrapper.find("#matchTeamTwo1").props().children).toBe(
      match[1].team2
    );
  });
  it("should have a match name in fixture ", () => {
    expect(wrapper.find("#matchName0")).toBeTruthy();
  });
  it("Check history push as a function ", () => {
    wrapper.find("#recent-match0").simulate("click");
    expect(history.push).toBeCalledWith(
      "/matchdetails/" + match[0].match_id,
      match[0]
    );
  });
  it("check recent match Data", () => {
    expect(wrapper.find("#matchData0").text()).toEqual("  - ODI");
  });
  it("Check history push as a function ", () => {
    wrapper.find("#recent-match1").simulate("click");
    expect(history.push).toBeCalledWith(
      "/matchdetails/" + match[1].match_id,
      match[1]
    );
  });
  it("Check function call to GetHeadline ", () => {
    wrapper.find("#getHeadlineData0").simulate("click");
    expect(getHeadline).toBeCalledWith(headlines[0].headline_id);
  });
});
