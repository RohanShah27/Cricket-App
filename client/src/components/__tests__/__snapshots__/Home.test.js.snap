// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Team Component should render the component 1`] = `
<Home
  fixtures={
    Array [
      Object {
        "match": "India vs Pakistan",
        "match_date": "29/11/1997",
        "time": "12:00 IST",
      },
      Object {
        "match": "India vs Srilanka",
        "match_date": "5/12/1997",
        "time": "12:00 IST",
      },
    ]
  }
  getFixtures={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getHeadline={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getHeadlines={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  getMatchesRecentMatches={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  headline={
    Array [
      Object {
        "headline_description": "test Descrption",
        "headlines": "Test headline",
      },
    ]
  }
  headlines={
    Array [
      Object {
        "headline_id": 1,
        "headlines": "Master Blaster Sachin baby",
      },
      Object {
        "headline_id": 2,
        "headlines": "BCCI President elections coming up",
      },
    ]
  }
  home={[MockFunction]}
  match={
    Array [
      Object {
        "match_competition": "Big bAsh leagues",
        "match_id": 2,
        "match_type": "ODI",
        "team1": "India",
        "team2": "Pakistan",
      },
      Object {
        "match_competition": "Big bAsh leagues",
        "match_id": 2,
        "match_type": "T20",
        "team1": "India",
        "team2": "Srilanka",
      },
    ]
  }
>
  <div
    style={
      Object {
        "marginTop": "80px",
      }
    }
  >
    <div
      className="home-flex-container"
      id="test"
    >
      <div
        className="site"
      >
        <div
          className="fixtures "
        >
          <h2
            className="fixtures-header"
            id="fixtures-header"
          >
            Fixtures
          </h2>
          <div
            className="col-content"
          >
            <ul
              className="home-ul"
            >
              <p
                class="home-p"
              >
                <span
                  id="matchName0"
                >
                  India vs Pakistan
                </span>
                 
                <span
                  className="home-time"
                >
                  29/11/1997
                   at 
                  12:00 IST
                </span>
                <hr
                  className="home-hr"
                />
              </p>
              <p
                class="home-p"
              >
                <span
                  id="matchName1"
                >
                  India vs Srilanka
                </span>
                 
                <span
                  className="home-time"
                >
                  5/12/1997
                   at 
                  12:00 IST
                </span>
                <hr
                  className="home-hr"
                />
              </p>
            </ul>
          </div>
        </div>
        <div
          className="news"
        >
          <div
            className="col-center"
          >
            <h2
              className="news-header"
              id="news-Section"
            >
              India's Tour of WestIndies
            </h2>
            <div
              col-content={true}
            >
              <figure>
                <img
                  className="homeimg"
                  src="DHONI.png"
                />
              </figure>
              <figcaption>
                <span
                  className="newsTitle"
                >
                  Test headline
                </span>
                <hr
                  className="home-hr"
                />
                <p
                  className="home-p"
                  style={
                    Object {
                      "fontSize": "15px",
                      "paddingLeft": "5px",
                    }
                  }
                />
              </figcaption>
            </div>
          </div>
        </div>
        <div
          className="headlines"
        >
          <h2
            className="fixtures-header"
            id="headlines"
          >
            Headlines
          </h2>
          <div
            className="col-content"
          >
            <ul
              className="home-ul"
            >
              <a
                onClick={[Function]}
              >
                 
                <li
                  className="home-li"
                  id="singleheadline0"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  Master Blaster Sachin baby
                </li>
              </a>
              <hr
                className="home-hr"
              />
              <a
                onClick={[Function]}
              >
                 
                <li
                  className="home-li"
                  id="singleheadline1"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  BCCI President elections coming up
                </li>
              </a>
              <hr
                className="home-hr"
              />
            </ul>
          </div>
        </div>
        <div
          className="home-match-container"
        >
          <div
            className="testimonials"
            id="#recent-matches"
          >
            <div
              className="card"
            >
              <span
                className="tournament-home"
              >
                 
                 
                - 
                ODI
              </span>
              <div
                className="parent"
              >
                <div
                  className="first"
                >
                  <img
                    className="imgformatch"
                    src="india.jpg"
                  />
                  <p
                    id="matchTeamOne0"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    India
                  </p>
                </div>
                <div
                  className="second"
                >
                  <button
                    className="buttonformatch"
                  >
                    View
                  </button>
                </div>
                <div
                  className="third"
                >
                  <img
                    className="imgformatch"
                    src="pakistan.jpg"
                  />
                  <p
                    id="matchTeamTwo0"
                  >
                    Pakistan
                  </p>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <span
                className="tournament-home"
              >
                 
                 
                - 
                T20
              </span>
              <div
                className="parent"
              >
                <div
                  className="first"
                >
                  <img
                    className="imgformatch"
                    src="india.jpg"
                  />
                  <p
                    id="matchTeamOne1"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    India
                  </p>
                </div>
                <div
                  className="second"
                >
                  <button
                    className="buttonformatch"
                  >
                    View
                  </button>
                </div>
                <div
                  className="third"
                >
                  <img
                    className="imgformatch"
                    src="pakistan.jpg"
                  />
                  <p
                    id="matchTeamTwo1"
                  >
                    Srilanka
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
  </div>
</Home>
`;
